linker.ld

```diff
@@ -1,30 +1,35 @@
 OUTPUT_ARCH(arm)
 MEMORY {
-        out     : ORIGIN = 0x23D7FF0, LENGTH = 0x8010
+        /* 
+                Overlay load address + offset to common area
+                see https://docs.google.com/document/d/1Rs4icdYtiM6KYnWxMkdlw7jpWrH7qw5v6LOfDWIiYho
+        */
+        main    : ORIGIN = 0x23D7FF0, LENGTH = 0x8010
+        /* Change/Add memory locations here */
+        /* NEW    : ORIGIN = 0x2000000, LENGTH = 0x800 */
 }
 
 SECTIONS {
-        .text : {
-
-                FILL (0xABCD)
-
-                __exidx_start = .;
-                __exidx_end = .;
-                abort = .;
-
-                __text_start = . ;
-
-                __init_array_start = .;
-                KEEP (*(SORT_BY_INIT_PRIORITY(.init_array.*)))
-                KEEP (*(.init_array))
-                __init_array_end = .;
-                __fini_array_start = .;
-                KEEP (*(SORT_BY_INIT_PRIORITY(.fini_array.*)))
-                KEEP (*(.fini_array))
-                __fini_array_end = .;
-                
+        /*
+                Add more sections for the linker, each section using a different memory location
+                This allows to spread your code on different overlays
+                You can change the section of one symbol (function or variable) in your C code using: __attribute__((section(".YOURSECTIONNAME")))
+                Note that section names are standardized and in the form ".text.OV.NB.SPECIAL"
+                Where OV is between 'ov9' (ARM9 Overlay) and 'arm' (Main ARM Binary)
+                NB is the OV number (7 or 9 for 'arm', any ARM9 Overlay number for 'ov9')
+                SPECIAL is a distinct name, this is only to distinguish between 2 sections on the same overlay
+        */
+        /*.text.OV.NB.SPECIAL : {
+                *(.YOURSECTIONNAME)
+        } >NEW = 0xff*/
+        /*
+                Main Section (Default section, in ARM9 overlay 36)
+                Note that you should add your special sections before that
+        */
+        .text.ov9.36.main : {
                 *(.init)
                 *(.text)
+                *(.text.*)
```

Comment:
What's the reason why this line was added? I'm not an expert in linker scripts, but it looks like the linker would include all `*.text.OV.NB.SPECIAL` sections again here (or does the linker never include the same section twice)?

Reply:
No particular reason, except to catch possible .text subsections that would not be included in properly defined sections in the final .elf otherwise.

`.text.OV.NB.SPECIAL` should not be included here because they are sections built by the linker for the final elf, sections caught within are from the object files linked (the linker script defines what sections from the input object files are in what section of the final .elf), and are not included twice.